main.cc

```diff
@@ -1032,6 +1034,22 @@ int main(int ac, char** av) {
                 repair_uninit_messaging_service_handler().get();
             });
 
+            supervisor::notify("starting CDC Generation Management service");
+            /* This service uses the system distributed keyspace.
+             * It will only do that *after* it has joined the token ring, and the token ring joining
+             * procedure (`storage_service::init_server`) is responsible for initializing sys_dist_ks.
+             *
+             * However, sys_dist_ks is stopped *before* CGM is stopped (`storage_service::drain_on_shutdown` below),
+             * so CGM takes sharded<db::sys_dist_ks> and must check local_is_initialized() every time it accesses it,
+             * then take local_shared() (this will prevent sys_dist_ks from being destroyed while CGM operates on it).
+             */
```

Comment:
Why did you decide to define cdc_gen_svc at the top instead of here? It seems that some services are defined at the top while others just before being started.

Reply:
because another service, `storage_service`, refers to it - and it's started before generation service